workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/SelectPaymentMethod/components/BillingAddress/BillingAddress.tsx

```diff
@@ -0,0 +1,244 @@
+		}
+	}, [billingAddress, setEnablePurchaseButton]);
+
+	return (
+		<Section
+			className="billing-address-section w-100"
+			label="Billing Address"
+		>
+			<div className="billing-address-section-card-addresses">
+				{addresses.map((address, i) => {
+					const {description, title} = getPostalAddressDescription(
+						address
+					);
+
+					return (
+						<RadioCard
+							description={description}
+							key={i}
+							onChange={() => {
+								setSelectedAddress(address.name as string);
+
+								const postalAddress = addresses.find(
+									(address) => address.name === title
+								);
+
+								const billingAddress: BillingAddress = {
+									city: postalAddress?.city,
+									country: postalAddress?.countryISOCode,
+									countryISOCode: 'US',
+									name: postalAddress?.name,
+									phoneNumber: postalAddress?.phoneNumber,
+									regionISOCode: postalAddress?.regionISOCode,
+									street1: postalAddress?.street1,
+									street2: postalAddress?.street2,
+									zip: postalAddress?.zip,
+								};
+
+								setShowNewAddressButton(false);
+
+								setBillingAddress(billingAddress);
+							}}
+							selected={selectedAddress === address.name}
+							title={title}
+						/>
+					);
+				})}
+			</div>
+
+			{showNewAddressButton ? (
+				<>
+					<button
+						className="align-items-center billing-address-section-card-new-address d-flex justify-content-center mt-4 w-100"
+						onClick={() => setShowNewAddressButton(false)}
+					>
+						<ClayIcon symbol="plus" />
+
+						<span>New Address</span>
+					</button>
+				</>
+			) : (
+				<div className="billing-address-section-card-container h-auto w-100">
+					<div className="align-items-center billing-address-section-card-header d-flex justify-content-between w-100">
+						<span className="billing-address-section-card-header-left-content">
+							New Address
+						</span>
+
+						<button
+							className="px-4 py-2"
+							onClick={() => {
+								setShowNewAddressButton(true);
+								setSelectedAddress('');
+
+								const billingAddress: BillingAddress = {
+									city: '',
+									country: '',
+									countryISOCode: 'US',
+									name: '',
+									phoneNumber: '',
+									regionISOCode: '',
+									street1: '',
+									street2: '',
+									zip: '',
+								};
+
+								setBillingAddress(billingAddress);
+							}}
+						>
+							Cancel
+						</button>
+					</div>
+
+					<div className="billing-address-section-container d-flex flex-column p-4 w-100">
+						<Input
+							label="Full Name"
+							onChange={({target}) => {
+								setBillingAddress({
+									...billingAddress,
+									name: target.value,
+								});
+							}}
+							required
+							value={billingAddress?.name}
+						/>
+
+						<Input
+							label="Address"
+							onChange={({target}) => {
+								setBillingAddress({
+									...billingAddress,
+									street1: target.value,
+								});
+							}}
```

Comment:
Please, remove the {} here.
```suggestion
							onChange={({target}) => 
								setBillingAddress({
									...billingAddress,
									street1: target.value,
								});
							}
```